src/registry/index.ts

```diff
@@ -0,0 +1,131 @@
+import { type SignerConfig } from "../helpers/connection.js";
+import { type ContractTransaction } from "../helpers/ethers.js";
+import { type TableIdentifier } from "./contract.js";
+import { listTables } from "./tables.js";
+import { safeTransferFrom, type TransferParams } from "./transfer.js";
+import {
+  setController,
+  type SetParams,
+  getController,
+  lockController,
+} from "./controller.js";
+import { createTable, type CreateTableParams } from "./create.js";
+import { runSQL, type RunSQLParams } from "./run.js";
+
+export {
+  type Result,
+  type Metadata,
+  type WaitableTransactionReceipt,
+  type Named,
+} from "./utils.js";
+
+export {
+  type TableIdentifier,
+  type CreateTableParams,
+  type RunSQLParams,
+  type TransferParams,
+  type SetParams,
+};
+
+export class Registry {
+  constructor(readonly config: SignerConfig) {}
+
+  async listTables(owner?: string): Promise<TableIdentifier[]> {
```

Comment:
This is missing any prefix information. I've tested out usage patterns based on this, and I think for the most part, this is fine. But we'll want to test that out in real life to be sure.